src/position.cpp

```diff
@@ -1017,7 +1017,7 @@ void Position::do_null_move(StateInfo& newSt) {
 
   sideToMove = ~sideToMove;
 
-  set_check_info(st);
+  set_check_info();
 
   st->repetition = 0;
```

Comment:
I wonder if this line `st->repetition = 0;` could be removed also, it generates the same bench for me, I would appreciate it if someone explained how though?

Reply:
I think even if we omit zeroing st->repetition on null move the bench remains stable because of the st->pliesFromNull guard which prevents to consider a move-sequence as 3-fold-repetition when there's a null-move in the chain.